lib/src/repo.rs

```diff
@@ -1032,6 +1032,40 @@ impl MutableRepo {
         new_ids
     }
 
+    /// Fully resolves transitive replacements in `parent_mapping`.
+    ///
+    /// If `parent_mapping` contains cycles, this function will panic.
+    fn resolve_rewrite_mapping_with(
+        &self,
+        mut predicate: impl FnMut(&Rewrite) -> bool,
```

Comment:
Does this function take a predicate because you expect to add another caller that will use a different predicate?